src/vm/clarity.rs

```diff
@@ -400,7 +453,11 @@ impl <'a> ClarityBlockConnection <'a> {
     fn with_abort_callback<F, A, R>(&mut self, to_do: F, abort_call_back: A) -> Result<(R, AssetMap, Vec<StacksTransactionEvent>), Error>
     where A: FnOnce(&AssetMap, &mut ClarityDatabase) -> bool,
           F: FnOnce(&mut OwnedEnvironment) -> Result<(R, AssetMap, Vec<StacksTransactionEvent>), Error> {
-        let mut db = ClarityDatabase::new(&mut self.datastore, &self.header_db);
+        let log = self.log.take()
+            .expect("BUG: Transaction Connection lost db log connection.");
```

Comment:
If we're worried that this `expect()` might trigger, maybe the act of `take()`ing and `replace()`ing the log could be done in a wrapper function that ran the in-between code as a closure?

Reply:
I kind of feel the same way about the `take()/replace()` usage with the cost tracker as well.

Reply:
Yeah -- that should be doable.

Reply:
Implemented in `e178739` -- it needs to use a macro, because attempting to do this across function boundaries steps on the borrow checker

Reply:
Looks good to me!